math/mathcore/inc/TMathVectorized.h

```diff
@@ -0,0 +1,350 @@
+// @(#)root/mathcore:$Id$
+// Author: Alejandro García Montoro 06/2017
+
+/*************************************************************************
+ * Copyright (C) 2017, CERN                                              *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifdef R__HAS_VECCORE
+
+#ifndef ROOT_TMathVectorized
+#define ROOT_TMathVectorized
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TMathVectorized                                                      //
+//                                                                      //
+// Encapsulate vectorized version of the most frequently used Math      //
+// functions.                                                           //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#include "Rtypes.h"
+#include "TError.h"
+#include "Math/Math_vectypes.hxx"
+//
+#include <algorithm>
+#include <limits>
+#include <cmath>
+#include <VecCore/VecCore>
+
+using ROOT::Double_v;
+using ROOT::Int_v;
+using ROOT::Float_v;
+
+using namespace vecCore;
+
+/* **************************************** */
+/* *         Base math functions          * */
+/* * Linear versions defined in TMathBase * */
+/* **************************************** */
+namespace TMath {
+
+   // TODO: VecCore types for short, int, long...?
+   // // Abs
+   // inline Short_t  Abs(Short_t d);
+   // inline Int_t    Abs(Int_t d);
+   // inline Long_t   Abs(Long_t d);
+   // inline Long64_t Abs(Long64_t d);
+   // inline Float_t  Abs(Float_t d);
+   // inline Double_t Abs(Double_t d);
+   // inline LongDouble_t Abs(LongDouble_t d);
```

Comment:
Yes, these comments have to go. The initial idea was to start vectorizing the commented functions, but a lot of them still remain.

Ok, I could try to template the functions instead of overloading them.